lisa/tests/scheduler/load_tracking.py

```diff
@@ -808,432 +825,13 @@ def make_group_bundle(cpu, item_group):
-class OneTaskCPUMigration(CPUMigrationBase):
-    """
-    Some tasks on two big CPUs, one of them migrates in its second phase.
-    """
-
-    @classmethod
-    def get_nr_required_cpu(cls, plat_info):
-        return 2
-
-    @classmethod
-    def _get_rtapp_profile(cls, plat_info):
-        cpus = cls.get_migration_cpus(plat_info)
-        nr_cpus = len(cpus)
-
-        periodic_settings = dict(
-            duration=cls.PHASE_DURATION,
-            period=cls.TASK_PERIOD,
-        )
-
-        return {
-            # A task that will migrate to another CPU
-            'migr': add(
-                RTAPhase(
-                    prop_wload=PeriodicWload(
-                        duty_cycle_pct=20,
-                        scale_for_cpu=cpu,
-                        **periodic_settings,
-                    ),
-                    prop_cpus=[cpu],
-                )
-                for cpu in cpus
-            ),
-            **{
-                # Just some tasks that won't move to get some background utilization
-                f"static{i}": nr_cpus * RTAPhase(
-                    prop_wload=PeriodicWload(
-                        duty_cycle_pct=30,
-                        scale_for_cpu=cpus[i],
-                        **periodic_settings,
-                    ),
-                    prop_cpus=[cpus[i]]
-                )
-                for i in range(min(2, nr_cpus))
-            }
-        }
-
-
-class NTasksCPUMigrationBase(CPUMigrationBase):
-    """
-    N tasks on N CPUs, with all the migration permutations.
-    """
-
-    @classmethod
-    def _get_rtapp_profile(cls, plat_info):
-        cpus = cls.get_migration_cpus(plat_info)
-        def make_name(i):
-            return f'migr{i}'
-        nr_tasks = len(cpus)
-        # Define one task per CPU, and create all the possible migrations by
-        # shuffling around these tasks
-        profile = {}
-        for cpus_combi in itertools.permutations(cpus, r=nr_tasks):
-            for i, cpu in enumerate(cpus_combi):
-                task_name = make_name(i)
-                task = profile.setdefault(task_name, RTAPhase())
-                profile[task_name] = task + RTAPhase(
-                    prop_wload=PeriodicWload(
-                        duty_cycle_pct=50,
-                        scale_for_cpu=cpu,
-                        duration=cls.PHASE_DURATION,
-                        period=cls.TASK_PERIOD,
-                    ),
-                    prop_cpus=[cpu],
-                )
-
-        return profile
-
-
-class TwoTasksCPUMigration(NTasksCPUMigrationBase):
-    """
-    Two tasks on two big CPUs, swap their CPU in the second phase
-    """
-    @classmethod
-    def get_nr_required_cpu(cls, plat_info):
-        return 2
-
-
-class NTasksCPUMigration(NTasksCPUMigrationBase):
-    """
-    N tasks on N CPUs, and try all permutations of tasks and CPUs.
-    """
-
-    @classmethod
-    def get_nr_required_cpu(cls, plat_info):
+class RqInvariance(InvarianceBase):
```

Comment:
nit: 2 blank lines between module-level declarations

Reply:
Will change it.